sdk/common/src/main/java/io/opentelemetry/sdk/resources/detectors/ServiceInstanceDetector.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.resources.detectors;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.sdk.resources.Entity;
+import io.opentelemetry.sdk.resources.EntityDetector;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+public final class ServiceInstanceDetector implements EntityDetector {
+  private static final String SCHEMA_URL = "https://opentelemetry.io/schemas/1.28.0";
+  private static final String ENTITY_TYPE = "service.instance";
```

Comment:
Are these entity types and their associated identifying / descriptive attributes codified anywhere yet?

Reply:
https://github.com/open-telemetry/semantic-conventions/tree/main/docs/resource

We're working on formalizing and stabilizing.  Starting with `k8s.*` here: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/k8s.md